Northstar.Custom/mod/scripts/vscripts/sh_northstar_http_requests.gnut

```diff
@@ -0,0 +1,235 @@
+globalize_all_functions
+
+global enum HttpRequestMethod
+{
+    GET = 0,
+    POST = 1
+    HEAD = 2,
+    PUT = 3,
+    DELETE = 4,
+    PATCH = 5,
+    OPTIONS = 6,
+}
+
+global struct HttpRequest
+{
+    /** Method used for this http request. */
+    int method
+    /** Base URL of this http request. */
+    string url
+    /** Headers used for this http request. Some may get overridden or ignored. */
+    table< string, array< string > > headers
+    /** Query parameters for this http request. */
+    table< string, array< string > > queryParameters
+    /** The content type of this http request. Defaults to application/json & UTF-8 charset. */
+    string contentType = "application/json; charset=utf-8"
+    /** The body of this http request. If set, will override queryParameters.*/
+    string body
+    /** The timeout for the http request in seconds. Must be between 1 and 60. */
+    int timeout = 60
+    /** If set, the override to use for the User-Agent header. */
+    string userAgent
+}
+
+global struct HttpRequestResponse
+{
+    /** The status code returned by the remote the call was made to. */
+    int statusCode
+    /** The body of the response. */
+    string body
+    /** The raw headers returned by the remote. */
+    string rawHeaders
+    /** A key -> values table of headers returned by the remote. */
+    table< string, array< string > > headers
+}
+
+global struct HttpRequestFailure
+{
+    /** The error code returned by native for this failure. */
+    int errorCode
+    /** The reason why this http request failed. */
+    string errorMessage
+}
+
+struct HttpRequestCallbacks
+{
+    /** 
+     * The function to call if the HTTP request was a success.
+     * Passes in the response received from the remote.
+     */
+    void functionref( HttpRequestResponse ) onSuccess
+
+    /**
+     * The function to call if the HTTP request failed.
+     */
+    void functionref( HttpRequestFailure ) onFailure
+}
+
+table< int, HttpRequestCallbacks > pendingCallbacks
```

Comment:
I think normally Respawn puts stuff like this into a singleton struct called "file". I'm personally fine without it at least while its only the single variable here, if more get added its probably worth doing the singleton struct